test/caPutJsonLogTest.cpp

```diff
@@ -812,6 +851,39 @@ void runTests(void *arg) {
            &charArray1[0], strlen(charArray1), &charArray2[0], strlen(charArray1), "DBF_CHAR array (lso) test");
 
 
+    // Test metadata
+    std::map<std::string, std::string> metadata_test_single = {
+        {"ABCDEFGHIJKLMNOPQRS", "1234567890_1234567890"}
+    };
+    std::map<std::string, std::string> metadata_test_escape_chars = {
+        {"ABC\n##&¤67&nbsp; space&#160&&#160;\'\\n#", "CBC\n##&¤67&nbsp; space&#160&&#160;\'\\n#"}
+    };
+    std::map<std::string, std::string> metadata_test_multiple = {
+        {"One", "1"}, {"Two", "2"}, {"Three", "3"}, {"Four", "4"}, {"Five", "5"},
+        {"Six", "6"}, {"Seven", "7"}, {"Eight", "8"}, {"Nine", "9"}, {"Ten", "10"},
+    };
+
+    std::map<std::string, std::string> metadata_test_reinsert_new_val = {
+        {"Five", "5"}, {"Six", "6"},
+    };
+
+    testMetadataHelper(metadata_test_single);
+    testMetadataHelper(metadata_test_escape_chars);
+    testMetadataHelper(metadata_test_multiple);
+
+    //Reinsert metadata test
+    logger->addMetadata("Five", "Five");
+    logger->addMetadata("Six", "Six");
+    testMetadataHelper(metadata_test_reinsert_new_val);
+
+    // 1000+ elements in metadata
+    std::map<std::string, std::string> metadata_test_big;
+    for(int i = 0; i <=1000; i++)
```

Comment:
Do we actually need to task 1000 inserts?

Reply:
Not really, but it actually caused an issue on the local json parser. The size of the buffer that receives the messages is 1024, with the 1000 inserts it caused to split the message. So in readFromClient, seems that incLogMsg.append(recvbuf) was appending the 1024 char but it continued until it found a '\n', so it was reading outside memory. I had to add recvLength to specify where to stop.

Reply:
100 elements would be enough IMO but if you want to keep 1000 it's fine.